packages/core/src/hydration/api.ts

```diff
@@ -130,6 +131,18 @@ export function withDomHydration(): EnvironmentProviders {
       useFactory: () => {
         let isEnabled = true;
         if (isBrowser()) {
+          // On the client, the DOM must contain a hydration marker or hydration is broken
+          const doc = getDocument();
+          const hydrationMarker = [...doc.body.childNodes].find(
+              node => node instanceof Comment && node.textContent === 'nghm');
```

Comment:
I think it'd be a bit better to use `nodeType` here and also extract the 'nghm' string into a const (so that we can also add JSDoc to it). We can put this const to `packages/core/src/hydration/utils.ts`.
```suggestion
              node => node.nodeType === Node.COMMENT_NODE && node.textContent === SSR_CONTENT_INTEGRITY_MARKER);
```